### PR TITLE
[deploy] Add git metadata to code locations

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

we should have been doing this sooner :shrug:

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* main
  * **PR #20** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster-cloud-action/20" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster-cloud-action/20?utm_source=stack-comment).

File: src/deploy.sh [hunk 1]
```diff
@@ -104,7 +104,9 @@ dagster-cloud workspace add-location \
     --location-name "${INPUT_LOCATION_NAME}" \
     --image "${INPUT_REGISTRY}:${INPUT_IMAGE_TAG}" \
     --location-load-timeout 600 \
-    --agent-heartbeat-timeout $AGENT_HEARTBEAT_TIMEOUT
+    --agent-heartbeat-timeout $AGENT_HEARTBEAT_TIMEOUT \
+    --git-url $BRANCH_URL \
```
Comment (prha) [inline src/deploy.sh:108]: this should be set regardless of whether or not this is branch deployment, yes?

**Review by prha — APPROVED** [review]

File: src/deploy.sh (hunk 1 above)
Comment (benpankow) [reply src/deploy.sh:108]: yup

Comment (benpankow) [reply src/deploy.sh:108]: nope

Comment (benpankow) [reply src/deploy.sh]: fixed

**Pull request merged**